### ISSUE TITLE
Prints start outside the bed instead of center where part is located

### ISSUE THREAD

**Issue by locutus40** [MarlinFirmware/Marlin#9772]
ok i upgraded to .18 added all my stuff.. i have a ramps and a arduino mega.. with prusa i3 and a bl touch...
it homes fine g29 fine.. when it goes to print it goes back to home then starts oozing out filament without moving at the home position... any ideas? i placed my print in the middle of the bed on slicer.. i also made sure i have 200 mm x 200mm set in marlin and repieter host... ... maybe a bad arduino? or driver going? ive had it 5 years...

heres a video..
http://www.mediafire.com/file/8fa0aun62lv5ama/20180223_014853.mp4

**Comment by thinkyhead** [comment]
Please test with the latest `bugfix-1.1.x` (and/or `bugfix-2.0.x`) branch to see where it stands. If the problem has been resolved then we can close this issue. If you still see the bad behavior we should investigate further.

**Comment by locutus40** [comment]
i am using bug fix 1.1x... it still does it.. and i cant use 2.0 because i use a mega which is not 32 bit.. i tried it again this time with slicer instead of prusa slicer.. same thing.. it shoots out filament like a faucet ... and just sits in home position...

**Comment by thinkyhead** [comment]
**Apologies if the "Issue" form was not prominent enough. We ask for certain details up-front so we don't just chase our tails. Please fill out the following form and one of our helpful volunteers will try to help solve your problem. Most importantly, we need to see your configurations.**

---
Thank you for submitting your feedback to the Marlin project.
Please use one of the templates below to fill out this box.

------------------------------------------------------------
### Feature Request
Please put [FR] in the issue title: `[FR] Add-on that goes 'ping'`

------------------------------------------------------------
### Compile Error
When I compile with `FEATURE_X` I get an error:
```
Paste_the_error_text_here
```

------------------------------------------------------------
### Bug Report
- Description: ---
- Expected behaviour: ---
- Actual behaviour: ---
- Steps to reproduce:
  - Do this
  - Do that

Attach a ZIP of `Configuration.h` and `Configuration_adv.h` by dropping here.

------------------------------------------------------------
### Bug Report Tips
- When troubleshooting, use `M502` followed by `M500` to reset EEPROM to defaults.
- Use `DEBUG_LEVELING_FEATURE` with `M111 S247` for detailed logging of homing/leveling.
- Format text with: **bold**, _italic_, `code`.
- Format C++ with three backticks, plus "cpp":
```cpp
void my_function(bool do_it) {
  // Hold this spot
}
```

**Comment by thinkyhead** [comment]
You'll have to re-post your video to YouTube. We have ethical issues with MediaFire.

**Comment by locutus40** [comment]
ethical issues? what did they do? im trying to figure out how to upload to youtube

**Comment by locutus40** [comment]
ok its uploading to youtube link shortly..

**Comment by thinkyhead** [comment]
> ethical issues? what did they do?

They're spammy, with hidden popups and the rest.
Meanwhile, with YouTube we don't have to download a 90MB file to see your issue.

**Comment by locutus40** [comment]
yea theres popups.. when you say ethical i thought your talking about them killing baby kittens or something.. pop ups are everywhere even on youtube.. but heres the youtube link..
i was also thinking i had a corruption of the boot loader so maybe its a failing arduino mega..
https://youtu.be/KgZIyzyFfxs

ive been sick for a month.. and life.... everywhere you look? ive posted an issue here and tried helping someone out about an hour ago...

i dont come on here often...reason being i havent had issues for a long time till i tried updating.. then it all went to hell fast

**Comment by thinkyhead** [comment]
> pop ups are everywhere even on youtube

Not actual new browser windows that appear when you
click a blank part of the background, they don't.
Mediafire does that. Like a porn or Torrent site.

> then it all went to hell fast

Ain't that always the way?

**Comment by thinkyhead** [comment]
So, to reiterate: ***Most importantly, we need to see your configuration files.***

Meanwhile, try turning off `SEGMENT_LEVELED_MOVES`. And reset your EEPROM with `M502` / `M500` just to be sure there's no funky old data lying around.

**Comment by locutus40** [comment]
where is segment leveled moves in config h? ill try all that now

**Comment by thinkyhead** [comment]
Where? Please use the search function of your text editor.

**Comment by locutus40** [comment]
ok ill have to go do this now.. and i dont know how to send you config.h files because i tried it says wont acept that file type..

**Comment by thinkyhead** [comment]
You can ZIP them up, as it explains in the template posted above.

_"Attach a ZIP of Configuration.h and Configuration_adv.h by dropping here."_

**Comment by locutus40** [comment]
ok might be tomorrow by the time i get all this figured out.. its 3 am here getting tired.. i m502 and m500 just now... im and im using arduino ide i see no text search function?

**Comment by locutus40** [comment]
[Configuration.zip](https://github.com/MarlinFirmware/Marlin/files/1751425/Configuration.zip)


hope this worked

**Comment by thinkyhead** [comment]
Try commenting out the lines `#define NUM_SERVOS 3` and `#define SERVO_DELAY { 300,300,300 }`. See if that helps at all.

**Comment by locutus40** [comment]
ok ill do that now.... i still cant fine this.. im using arduino ide theres no search function.. and debug turn on where is that in the config h file so i can turn it on?

**Comment by thinkyhead** [comment]
It's highly recommended to use a real text editor like Sublime Text or Notepad++ that contain the kind of search function we are accustomed to as higher primates, rather than fight the Arduino IDE.

**Comment by locutus40** [comment]
im dowloading notepad now.. ive disabled the servos still need to find that feature and go from there

**Comment by locutus40** [comment]
ok text editor is soo nice i disabled servos enabled debugging and disabled that other thing im going to upload heat up my bed and try a print stand by

**Comment by AnHardt** [comment]
Like in all cases where an axis is not moving, check the endstop states with M119.

**Comment by locutus40** [comment]
end stops work it homes and bed levels fine it just starts printing at home postiion and shoots out a ton of filament...  standbye heating now

**Comment by thinkyhead** [comment]
If the problem persists, also zip up the G-code file you're printing so we can see if any odd settings are being applied.

**Comment by locutus40** [comment]
ok its heating getting ready to home etc

**Comment by locutus40** [comment]
still did it.. its shooting out filament on the home position like mad so much so it cant keep up with it entering and its grinding.. ill upload a 200mb video to youtube its going to take a while so ill come back tomorrow and post it... here... im thinking its the arduino. but heres the new config.h file to look over to prove i did what you said i dont know how to get the debug thing so tell me how to do that..

[Configuration.zip](https://github.com/MarlinFirmware/Marlin/files/1751495/Configuration.zip)

theres the new config.. i show in the video i took where its printing in the exact center of the plate...
[velosynth-spoke-magnet-v1 (repaired).zip](https://github.com/MarlinFirmware/Marlin/files/1751499/velosynth-spoke-magnet-v1.repaired.zip)

theres the file im trying to print.. its a magnet holder for a bicycle spedo ive repaired it.. etc..

im at a loss for words guys... like i said it might be the arduinio its self seeing how the boot loader got corupted.. i had to upload a new bootloader via another arduino.. so it might be the chip its self...

take note i had to change sides of my x home to the left instead of the right when i used right after the error happened with the boot loader the x woulndt move no matter what i did so i placed it on the left and voila it works.. its all seeming like a hardware problem.. 5 years old.. something bound to break.. i just dont have another one to test or any money right now.. so if i get another one in a little bit a week or so ill re do it and test.. if no one else has any new input?

**Comment by thinkyhead** [comment]
I can't do anything with an STL file. We need to see the G-code, please.

**Comment by locutus40** [comment]
the g code hmm hold on let me get it

**Comment by thinkyhead** [comment]
Printing from a host or from SD Card?

**Comment by thinkyhead** [comment]
We'd like to see your `Configuration_adv.h` too, please.

**Comment by locutus40** [comment]
printing from sd  card or my pc it acts the same spews out filament etc.. im uploading the video to drop box 10 minute to go then i have to upload it to youtube it will take an hour for that.. . i did give you my config heres it again 
[Configuration.zip](https://github.com/MarlinFirmware/Marlin/files/1751525/Configuration.zip)

**Comment by locutus40** [comment]
oh wait advanced sorry lol

**Comment by locutus40** [comment]
[Configuration_adv.zip](https://github.com/MarlinFirmware/Marlin/files/1751528/Configuration_adv.zip)

**Comment by thinkyhead** [comment]
So, your XY homing position is where? On the front-left or the back-right?

**Comment by locutus40** [comment]
when looking at the machine aka video to the left and bed back towards me.... heres the gcode
[gcodeorigianl.zip](https://github.com/MarlinFirmware/Marlin/files/1751539/gcodeorigianl.zip)

**Comment by thinkyhead** [comment]
That G-code would really help too, whenever you can get a hold of it…

**Comment by thinkyhead** [comment]
That file only contains the very last bit of the G-code. Please export all the G-code from your slicer or host and send us that.

**Comment by locutus40** [comment]
i tried thats all that it gives me... let me try again

**Comment by thinkyhead** [comment]
Which slicer or host? They usually have an "Export G-code" function.

**Comment by locutus40** [comment]
[gcodefinal.zip](https://github.com/MarlinFirmware/Marlin/files/1751565/gcodefinal.zip)

slic3er and repiteir host.. i dont like pronterface because if you click a move button it can go way past your end stops hanvent figured out how to stop that so i just use repiteier all are up to date

**Comment by locutus40** [comment]
in about 20 minutes ill have a new youtube video of it doing the same thingwith the stuff you told me to turn off and on

**Comment by thinkyhead** [comment]
No need, thanks.

**Comment by thinkyhead** [comment]
Here's the first move in that G-code file. See anything interesting?
```gcode
G1 X-66.903 Y-66.983
```

**Comment by locutus40** [comment]
ok well its uploading...

so any thoughts? im like figuring a new board... or a new 32 bit board... ive never seen this problem before its strange... funny thing is its a genuine mega board

**Comment by thinkyhead** [comment]
Here's the first move in that G-code file. See anything interesting?
```gcode
G1 X-66.903 Y-66.983
```

**Comment by locutus40** [comment]
yea its placing the first move at the start position.. heres my work flow..

get part or design.. place in repiterhost.. then slice.. then hit print..  slicer is the one thats doing this then?

**Comment by locutus40** [comment]
[Uploading Screenshot 2018-02-23 04.17.11.zip…]()
 maybe its this.. notice the box is in the center of the bed.. heres my config for bed size on repiteier host..

im at a loss why would slicer tell it to start there.. maybe i have my bed set up wrong? i dont know

**Comment by thinkyhead** [comment]
> slicer is the one thats doing this then?

Yes, fix your slicer settings. -66, -66 is way off the bed. Tell the slicer that your bed goes from 0,0 to 200,200 and be careful to put the model in the middle before you export the g-code or print from it.

If your homing position is at the front-left corner of the bed, moving Y+ should bring the bed forward and moving X+ should move the nozzle to the right. If your home is at the back-right of the bed, your  Y+ should move the bed backward and X+ should move the nozzle left. If the machine is set up with the home in the front-right or the back-left of the bed, your handedness is backwards and objects will be printed mirrored.

**Comment by locutus40** [comment]
ok where in slicer do i make the changes you suggest?

**Comment by thinkyhead** [comment]
That's really beyond the scope of what we can do for you. Please continue with Google or http://forums.reprap.org/

**Comment by locutus40** [comment]
in bed shape i have x155 y155
origins x 0 y 0.. 

do i just put where y is 200,200? im freaking lost

[newscreenshot.zip](https://github.com/MarlinFirmware/Marlin/files/1751632/newscreenshot.zip)
see what i mean here..
you suggest telling my slicer where the bed is but cant tell me where to change the settings?

**Comment by locutus40** [comment]
https://youtu.be/6g9qFN5L2Eo heres what it does the new video same but with the whole process

**Comment by locutus40** [comment]
ok i just put in 200x 200 instead of 155x 155... and sliced it it still says -66 -66 as the first line.. so it did nothing... im going to delete slicer and try another program or download the new version if there is one... or just toss my printer and pc out the window ll

**Comment by locutus40** [comment]
yea im using 1.30 a dev version stable is 1.2.9 im gunna try that maybe thats why

**Comment by locutus40** [comment]
wth you closed it.. its not resolved... what if i do get a new board and its still doing it with fresh software etc then it is a bug.. leave it open

**Comment by thinkyhead** [comment]
It's not a bug. The problem is obvious to us with years in the field.

**Comment by locutus40** [comment]
ok its obvious to you.. you told me to do 0,0 200 200.. 

where ? in slicer whre do i do that?
btw i deleted everything downloaded fresh stuff
to make sure..

**Comment by locutus40** [comment]
nevermind im going to figure this out on my own...

**Comment by thinkyhead** [comment]
Thank you. We have many things to do, and this project repository is not a support site. This is a hub for Marlin development. For help from other users, http://forums.reprap.org/

**Comment by locutus40** [comment]
not a support site lol yet theres 500 issues with the firmware.. thats support or asking for help which i was doing.. i came up with a issue... and i belive someone else had a issue i was talking to similar like 2 days ago .. so to say you have many things to do.. implies your tooo important or members are too important to help little old me.. .. i came to you with a genuine issue.. ... if i do all this stuff and believe me ive got fresh copies of repiter host and the last stable version of slicer and it gives me the same issue then its a hardware... or your firmware and if i buy a new board and it still exsists its a firmware issue.. hence why i came here in the first place.. you know this is how you diag stuff is ask questions.. im a ase certified tech ive been working on electric vehicles an 5th wheels cars trucks you name it for over 20 years.. i ask questions still with all that knowledge i even have a college degree in my feild.. went though a physics class to get my degree in engineering and repair... so when im treated like this.. for asking questions it rolls me.. ive had a issue about ohh a few months ago hence why you asked me where i been because 3 of you got on my case  about it.. same treatment... i come here for help seems i end up fixing my own issues on my own.. and no help comes only critizizim god i suck at spelling that and its 5 am an di havent sleep in 24 hours...

so cut me some slack.. its a genuin issue and im trying to figure out what it is ..... thats all.. so dont chastize me for asking for alittle help or outside knowledge...

**Comment by thinkyhead** [comment]
Best of luck to you. Glad I could help solve the issue. Your happiness is important to us.

**Comment by AnHardt** [comment]
> not a support site lol yet theres 500 issues with the firmware.. th - bla bla bla

We (being able to read Marlins source) are able to find answers about everything you can ask about Marlin. Sometimes it needs a while, but in theory, we can.

The problem with your last question is - it's not about Marlin, but about your host program.

For RH: Ope4n 'Printer Settings', 'Printer Shape', check 'Classic Printer', - edit to your needs, 'OK'.

For Slic3r. Open 'Printer Settings' tab, click 'Bed shape:' 'Set...' , Edit Origin, 'OK'.

For Marlin play with:
```
// The size of the print bed
#define X_BED_SIZE 200
#define Y_BED_SIZE 200

// Travel limits (mm) after homing, corresponding to endstop positions.
#define X_MIN_POS 0
#define Y_MIN_POS 0
#define Z_MIN_POS 0
#define X_MAX_POS X_BED_SIZE
#define Y_MAX_POS Y_BED_SIZE
#define Z_MAX_POS 200
```

**Comment by locutus40** [comment]
// @section homing

// The center of the bed is at (X=0, Y=0)
//#define BED_CENTER_AT_0_0

what about this... if my bed is 200 mm x200mm what do i put in the above?

**Comment by AnHardt** [comment]
You can't put something in there. You can activate it, or not.
Sorry for forgetting about that.

**Comment by locutus40** [comment]
ok i just replaced all the stepper drivers on the ramps with brand new ones... voltages are 0.400 or 800 mah for all motors it doubles on the output side..

anyway it still does it.. ive uploaded the 1.18 firmware bed 200x200 mm did the same for repiter host...
after deleting repiter host and slicer and the folders and re downloading them both.. configured them 

re entered my printer specs bed 200 x200 by 180  in repiter and slicer take into account repiter host has slicer, prusa slicer edition and cura... i tried all 3.. i noticed slicer added a z hight to it in the first line so i said hmm let me try prusa same thing.. so i erased it since the firm ware has my -2.70 hight after homing i even g1 z0 and it puts the nozzel on the bed..
then i even tried cura.. didnt add any nozzel hight in the code.. but still i get the same stinking thing it was doing before..

well almost it prints now in the middle far right so not in the middle.. and prints 10 mm above the bed... so techinally the same thing.. the only thing i didnt replace is the arduino board.. im getting to that part.. 

so if its not the arduino when i replace it.. does any one annnyyone have any ideas??????

manually type in g1 z0 and it goes to the setting i set up 2 years ago right on the bed where it suppose to.. heres a youtube video of what it does ....

also i used a brand new stl file a calibration cube.. so i know its not the model...
i let repiter host place it in th middle and same effect far right and 10mm above... also if i move the middle all the way to the left of the screen bed same thing prints far right and 10mm above.. i even get a g28 error so i have to pre home each axis first before the slicer program does....

im ripping my hair out..
also you closed this wayy to early.. the problem or issue was not solved..
heres the vid
....https://youtu.be/zz7d2hT8t-Y
remember before it printed at home and out side the bed.. well i was using the fix version.. then went to what it does now... it might be hardware.. i did have a corruption of the boot loader and i did replace it... but who knows..

**Comment by thinkyhead** [comment]
- Be sure to reset EEPROM with `M502` and `M500` after any re-flash of Marlin.
- Be sure to test with the latest `bugfix-1.1.x` branch.
- It may help to also post your questions in the [Marlin Forum](http://forums.reprap.org/list.php?415) and the [Marlin Facebook Group](https://www.facebook.com/groups/1049718498464482/).
- Please ZIP up and attach:
  - The G-code you're having trouble with, and
  -  Your current configuration files.

**Comment by locutus40** [comment]
im about ready to toss the towl in and go with a another board a smoothie board with their firmware.... should eliminate my issues... thanks though

**Comment by thinkyhead** [comment]
Cheers, thanks for stopping by.

**Comment by locutus40** [comment]
who is the one that actually created this firmware? so i can talk to them? from what i understand its not you thinkyhead its 2 other guys

**Comment by thinkyhead** [comment]
Marlin Firmware is the joint effort of many, many volunteers. My job here is mainly custodial, incorporating submissions, investigating and fixing bugs, and occasionally helping to integrate new features.

One of the primary originators was Erik van der Zalm, currently an employee of Ultimaker, I believe. I spoke to him just today, and he does plan to make some contributions when he gains more free time.

**Comment by Roxy-3D** [comment]
> who is the one that actually created this firmware? so i can talk to them? from what i understand its not you thinkyhead its 2 other guys

It's a lot of people.   But Erik Zalm's name is put on any new file that is created even though he didn't create the file.   And the code he left us with pays homage to Sailfish and others.    This software has been evolving continuously for a long time.

> who is the one that actually created this firmware?

So my suggestion is you go hunt down the Sailfish and maybe Gerble people.

**Comment by thinkyhead** [comment]
Well, that's true. Marlin was a "mashup of Sprinter and Grbl" and sometimes I think it would be cool to bring in the planner from the latest Grbl as a kind of "refresh." They've made some great improvements.

**Comment by thinkyhead** [comment]
By the way, I wrote up and posted all those helpful docs we're supposed to have to let users know how this whole Issue Queue thing works, how to file bugs, and where users can get support if they're having trouble figuring out configuration. Have a look-see! https://github.com/MarlinFirmware/Marlin/blob/1.1.x/.github/contributing.md

**Comment by locutus40** [comment]
well what i will do is buy a new ramps and arduino its 20 bucks if that doesnt fix it im only 20 bucks into it and if it still.. does it.. im not sure what controller to get.. see marlin is easy it configures in arduino ide... im soo used to it...

and im not sure what other firmwares are out there that have the same features?

other than buying a whole new printer with some proprietary firmware and i just feed it gcode but thats another 300 plus dollars rather not drop that much.. thats y im pulling out my hair lol.. yea thats what i thought sojmeone else wrote it a mash up of grlb

**Comment by locutus40** [comment]
nice.. just looked..

yea i wish my coders could contribute to this.. i dont know code or very much of it to contribute..

see i was part of.. the founding fathers of carters addon pack.. a stargate addon pack for gmod the best one and only one out there with everything.. i was the sound engineer.. we did it for 5 years.. and let me tell you we have one good coder alex  and there was 2 others but they left.. i wish they would step on up but there not into 3d printing like me... hell of coders too im talking about stuff you see in stargate they coded for props.. working gates you name it.. talented people.. im not in touch with them anymore other wise id convince them to come on over im in touch with alex but hes not into this stuff

**Comment by Roxy-3D** [comment]
My mistake...   (I'm not detail orientated!)    At: reprap.org/wiki/List_of_Firmware#Marlin

> Marlin
> FIRMWARE INFO
> Details	Description
> Name : Marlin
> Author(s) : Erik van der Zalm: Active as of February 2014; Bernhard Kubicek: Active as of November 2011
> Development on this firmware appears to be very active. Forked from Sprinter and Grbl.
> Status : Active as of June 2014

It was Sprinter and not Sailfish....

**Comment by thinkyhead** [comment]
Nice! I used to write Amiga games, but left the industry in 1991 to drop acid and read philosophy books.

**Comment by locutus40** [comment]
in fact heres out steam thing.. im locutus_1 one of the 5 founders hold on let me find our page on steam

**Comment by locutus40** [comment]
https://steamcommunity.com/sharedfiles/filedetails/?id=180077636

Rafael De Jongh and i locutus_1 decided 5 years ago.. to start this pack seeing how no one had this stuff... alex is like you now a maintainer and fixer of the current code... we had 2 others before him..anyway im in the list there somewhere lol i stopped on the pack about 2 years ago.. its now since stolid just maintained we even had a website not anymore to expensive.. but anyway those guys could code.. let me tell you alex is pretty good...

**Comment by locutus40** [comment]
amiga i remember those games.. i go way back to atari and the old 80s arcades as a kid id drop 5 dollars back then in quarters thats like 10 dollars or 15 today...

**Comment by thinkyhead** [comment]
My first computer was a Sinclair ZX-80. My second was an Atari 400. I quickly replaced the membrane keyboard with a proper clicky one. I wrote a lot of cool things in AtariBASIC and 6502 assembler which I tried to get published in Softside and Compute! and I wish I still had that stuff. But all I have now is a disk edit and repair tool that someone happened to have a copy of and sent me 10 years ago. I still occasionally get "retro fever" and one of these days I'll publish a Defender clone I've written in Unity 3D.

**Comment by locutus40** [comment]
nice.. my first was the commador 64.. then apple II  and for home arcade atari 2600 ....now im dating myself haha

a defender clone you say... very nice...great now i want to play some of the old games haha

**Comment by thinkyhead** [comment]
Thank the gods for emulation.

**Comment by locutus40** [comment]
yea...cough cough M.a.m.e lol in fact i built a mame machine never used it hardly its set up for light gun games driving you name it has all that uk guys hardware that he made for keyboard emulation and stuff track ball i forget his name but yea now you can emulate in a window on your pc without the bulky roms and hard disc space...still fun dropping quarters in a mame machine though.. shoot i even had pinball on it.. something like 10,000 roms 3000 pinball tables you talk about firmware set up.. that was a nightmare... but yea alot of smart coders out there from 20 years ago all vanished.. its all new blood now face book etc... the old school guys are what we need back steve wolzniac .. forget how you spell his name.. is a god to me lol steve jobs can chew on a egg though haha

**Comment by thinkyhead** [comment]
I once threw a Galaga machine off a balcony. And then Gene Rayburn walked into the room.

**Comment by thinkyhead** [comment]
From your video @locutus40 it looks like you're tantalizingly close to having a working setup. One of the trickier things to configure is the Z probe Z offset. We want to add a guided procedure to get that tuned from the LCD. I will take a crack at that later in the week. In the meantime, could you post a video of just your `G28` and show how the Z position ends up, either on the LCD or with `M114`…?

**Comment by locutus40** [comment]
yea the z offset is set already
56
15
-2.70 
when i g1 z0 from my host it puts the nozzel right on the bed.. then when i slice its 10mm above the bed.. even with the first command line just cordinantes and not z offset.. so something is overriding or not telling the arduino that it needs to move down and over.. even when i place the work piece anywhere on the bed with cura with slicer with slicer prusa edition it still does not move it over prints 10mm above bed and to the right  so.. im like ready to toss this out the window..
reason im 50 50 on the issue is i have new steppers installed and configed right.. all new software.... so its a mystery..
where my g28 ends up is after homing i set it to move up 15mm so it probes then goes up... then when you print it goes down alittle and oozes stuff out right there.. ive fixed the massive ozzing problem it was a setting so i did get one thing i had 80mm a second instead of 5... or was it 2 .. i forget..  but it probs just fine..

now when i g28 from repiter host i get a g28 error and it stops once i try it again its fine.. so what im going to do is buy another ramps board and arduino it might be the arduino  i have an uno but i cant hook that up not enough ports.. so im going to drop 20 bucks next week and try that and if it still does it.. then its confirme to be the firmware....something is a miss lol....yea but g1 z0 drops it down to the bed i can fit paper between it like normal but print and its way up there.. something either in the code is not telling it to drop to the -2.70 in fact heres my config file take a look im on 1.1.8 now

[Configuration.zip](https://github.com/MarlinFirmware/Marlin/files/1778291/Configuration.zip)

**Comment by locutus40** [comment]
let me record and upload a youtube video to show you where it ends up

**Comment by locutus40** [comment]
i recorded a video now i have to upload to drop box then to youtube it will take a bit

**Comment by thinkyhead** [comment]
Currently your configuration says:
```cpp
#define Z_PROBE_OFFSET_FROM_EXTRUDER 2.70
```
Change that to -2.7. You can do it without re-flashing by using `M851 Z-2.7`. After you change it you'll have to do a new `G28`, `G29`, `M500`.

**Comment by thinkyhead** [comment]
Note number two:
```cpp
#define X_PROBE_OFFSET_FROM_EXTRUDER 56
```
This is also incorrect. With your probe on the left of the nozzle, this value should be -56. You **_will_** have to re-flash after changing that.

**Comment by locutus40** [comment]
ok ill try the positive 2.70.

see looking at my printer twhere the lcds are acording to the diagramin the firmware its the back of the printer so the left is on the right side.. and 56 is 56 mm on the right side of my head.. what your suggesting is i change that to negative.. ok let me go back in re flash and try that.. btw those numbers have been on the firmware since 1.02 way back when 2 years ago.. and theyve worked then but ill try anything now lol hold on

**Comment by thinkyhead** [comment]
Well, I'm assuming your home position is in the front-left corner of the bed. As I recall you may have said that your home position was the back-right of the bed. In which case, please disregard.

**Comment by thinkyhead** [comment]
Yeah, I was assuming your home position is in the **front-left** corner of the bed. I do recall you already said that your home position was the **back-right** of the bed.

That being the case, please disregard the `X_PROBE_OFFSET_FROM_EXTRUDER` note.

(I find it easier to configure when the front-left of the bed is used.)

**Comment by thinkyhead** [comment]
> ok ill try the positive 2.70.

Negative!

**Comment by locutus40** [comment]
where you saw my probe to the left and back.. so if your looking at the p rinter... left side and back...hold on changing numbers now i have to redo the probe positions

**Comment by locutus40** [comment]
wait so leave the 56 positive?

**Comment by thinkyhead** [comment]
When the home position is the left-rear of the bed or the front-right, either the X and Y axes must be swapped, or objects will be printed mirrored. Marlin requires right-handed coordinates.

**Comment by locutus40** [comment]
ok  its now 56 15 2.70 let me upload.. and try again i have to heat the bed up etc

**Comment by locutus40** [comment]
hold on ill show you where the end stop is taking a pic

**Comment by thinkyhead** [comment]
Negative! `M851 Z-2.7`

**Comment by locutus40** [comment]
ok ill redo it lol hold on

**Comment by thinkyhead** [comment]
https://drmrehorst.blogspot.com/2017/08/setting-up-3d-printers-origin.html

**Comment by locutus40** [comment]
hmm now its taking forever to upload now ok it took im heating now

**Comment by locutus40** [comment]
[Configuration.zip](https://github.com/MarlinFirmware/Marlin/files/1778299/Configuration.zip)

this is what it is now that i uploaded.. almost done heating

**Comment by thinkyhead** [comment]
Don't forget that `M502` is required to apply the new settings. `M500` to retain them on the next boot.

**Comment by locutus40** [comment]
done and stored.. now to slice and try to print

**Comment by locutus40** [comment]
stored the settings g28 now then 29 and it should print crosses fingers

**Comment by thinkyhead** [comment]
Do you include a `G28` and `G29` in all your prints? Once you get the mesh stabilized and saved to EEPROM, you can then just use `M420 S1` and skip the probing. (If you feel confident that the bed won't move.)

**Comment by locutus40** [comment]
yes i g28 and g29 every slice its in the slicer software to make sure it does it every time

**Comment by locutus40** [comment]
bad news it still prints 10mm above same as the first video i showed you and far to the right

**Comment by locutus40** [comment]
i had to emergency stop.. i did the m500 and save settings command.. already then g28 and 29 after that and before.. i just dont get it...

**Comment by locutus40** [comment]
oh and -2.70 it was already on that when i g1 z0 it goes right down to the bed..... is there any other slicer firmware out there? other than cura and slic3er.. thats free.. i might try another program  what about another host to move the printer instead of repiter?  im trying to cover all angles

**Comment by locutus40** [comment]
im going to try mattercontrol software it has a built in slicer.. just to rule out the programs..

**Comment by locutus40** [comment]
ughh it dont work cant connect so im off to find a host software

**Comment by thinkyhead** [comment]
Try this:

- Download Marlin `bugfix-1.1.x` to test with the latest code, using your 1.1.8 configs.
- Enable `DEBUG_LEVELING_FEATURE` and `M114_DETAIL` and re-flash the firmware.
- Connect to your printer from host software such as Cura, Printrun or Repetier Host.
- Issue the command `M111 S247` to enable maximum logging.
- Perform a `G28` to do your standard homing procedure.
- Do a `G29` to probe the bed. This will also enable bed leveling.
- Copy the log output into a **.TXT file** and attach it to your next reply.

With bed leveling enabled you can use `M114 D` as you move around the bed to see lots of details about the current position. If you see strange things, also copy them into your text file.

Since you're using `BILINEAR` leveling, you can also use `G29 Q` at any time to see details about leveling at the current position.

I look forward to your next report! Hopefully we can narrow this down and fix it!

**Comment by locutus40** [comment]
first im trying printrun as a host and another slicer..... if it still does it then ill try the above

**Comment by thinkyhead** [comment]
I've found that Printrun is the most bare-bones host which makes it the easiest to get a complete unmolested log output, so I tend to use Printrun when testing.

**Comment by locutus40** [comment]
ok heres what im trying repiter host.. still.. and i bought simplify 3d just now.. maybe its the slicer.. i tried printrun host it didnt like my setup it still uses slicer i dont want slicer lol so im going to just use repiter to control since i know it and simply3d to slice and try that.. then if that dont work theres 2 more sliceers i can try kiss and something else..  ughhhh i wanna toss this thing otu the window haha im sooo frustrated funny thing is i have a cheap 180 delta printer to the left of this printer it works fine it uses a propriatary 32 bit  controller and a cheap easy interface and poof it works i use cura for it but i need this one to so i can do nylon and other filaments.. so let me try this stuff ill get back to you

**Comment by thinkyhead** [comment]
Ok, I'm off to bed. I'll check back in the daylight hours. Good luck!

**Comment by locutus40** [comment]
ok i just used a whole new program simplfy3d sliced it with it... set it all up even typed into the slicer G1 Z0 after the G29 command.. still it prints mid air and the same spot..... so next step is toss this pile out the window lol.. no next step is drop 20 bucks and get a whole new controller.. that will be next week ill let you know

**Comment by thinkyhead** [comment]
Sorry to proselytize, but as an expert in this field, I would not be replacing the controller at this stage. I would be investigating my settings and examining key data that reveals what is going on:
- The G-code you're using. **We very much need to see this first!**
- The output of `M503 S0`, so we know your active settings.
- The details of your homing and probing procedure. To wit:

Whenever there are homing or leveling issues, we ask users to follow a specific procedure to gather more information so that we're not just taking stabs in the dark. Here is the boilerplate:

- Download Marlin `bugfix-1.1.x` or `bugfix-2.0.x` to test with the latest code.
- Enable `DEBUG_LEVELING_FEATURE` and `M114_DETAIL` and re-flash the firmware.
- Connect to your printer from host software such as Cura, Printrun or Repetier Host.
- Issue the command `M111 S247` to enable maximum logging.
- Perform a `G28` to do your standard homing procedure.
- Do a `G29` to probe the bed. This also enables bed leveling.
- Do some of the moves that revealed problems before. Take notes.
- Copy the log output into a **.TXT file** and attach it to your next reply.

Repeat this procedure, if needed, to demonstrate inconsistencies. From these logs we should hopefully get a better idea of what's going on with your machine.

You can use `M114 D` as you move around the bed to see lots of details about the current position. If you see strange things, also copy them into your text file.

Since you're using `BILINEAR` leveling, you can also use `G29 Q` at any time to see details about leveling at the current position.

It would be good to have this information and apply the Scientific Method. From that it would be possible to narrow this down to a single cause and get you a working setup.

**Comment by locutus40** [comment]
ok i scrounged some money together did a few mafia hits here and there... lol jk and it should be here within a week then well see...

**Comment by locutus40** [comment]
ok i got my ramps and arduino and brand new steppers.. it does the same thing.. prints 10 mm above the bed even when i put in manually the offset in the first line it wont move down....

also the touch screen mks i have does not work.. plugged in same way as my other stuff..

so anyone want a printer? im so fed up... i dont get it it makes noooooooooooooo sense...

steppers are plugged in right they move where there suppose to up and down with marlin  so im giving up... im selling it for parts.. even the ive tried every slicer tried every single one..

**Comment by thinkyhead** [comment]
> it does the same thing

Yes, well the electronics are not in any way implicated by your symptoms.

> so anyone want a printer? im so fed up... i dont get it it makes noooooooooooooo sense...

Is there no 3D printing club in your local area which could help you in person? If not, you may be able to find someone nearby who you can ship it to. They could get everything set up properly for you and then ship it back. Ask around the social networks.

If you do decide to give up utterly, we are always needing donations of equipment to help in development.
